Test_Engine/Compute/ToJsonWithChecks.cs

```diff
@@ -0,0 +1,143 @@
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Diffing;
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Test
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Runs through the provided objects and tries to convert to Json. Then performs checks if the object can be deserialised, and if the de-serailised object is equal to the serialised version.")]
+        [Input("objects", "Objects to try to convert ToJson.")]
+        [MultiOutput(0, "validJson", "Json strings that pass all checks.")]
+        [MultiOutput(1, "failingToJson", "Json strings show some failure of running the ToJson convert.")]
+        [MultiOutput(2, "failingToJsonObject", "Obejcts that failed the ToJson convert.")]
+        [MultiOutput(3, "failingFromJson", "Json for cases where the FromJson is failing.")]
+        [MultiOutput(4, "failingFromJsonObjects", "Objects that had some failures going FromJson.")]
+        [MultiOutput(5, "notEqualJson", "Jsons strings for objects not equal to incoming object after serialisation ToJson followed by deserialisation FromJson.")]
+        [MultiOutput(6, "notEqualObjects", "Objects not equal to incoming object after serialisation ToJson followed by deserialisation FromJson.")]
+        public static Output<List<string>, List<string>, List<object>, List<string>, List<object>, List<string>, List<object>> TryToJsonAndCheckIfEqual(List<object> objects)
+        {
+            List<string> successfullJson = new List<string>();
+            List<string> failingToJson = new List<string>();
+            List<object> failingToJsonObjects = new List<object>();
+            List<string> failingFromJson = new List<string>();
+            List<object> failingFromJsonObjects = new List<object>();
+            List<string> notEqualJson = new List<string>();
+            List<object> notEqualObjects = new List<object>();
+
+
+            foreach (object obj in objects)
+            {
+                string json;
+                try
+                {
+                    json = BH.Engine.Serialiser.Convert.ToJson(obj);
+                }
+                catch (Exception e)
+                {
+                    failingToJson.Add("");
+                    failingToJsonObjects.Add(obj);
+                    BH.Engine.Base.Compute.RecordError(e, $"Failed ToJson for {obj.GetType()}");
```

Comment:
Does this error bring any new meaning compared to the log? If not, I think I would skip it not to double the information

Reply:
Was omstly in case the exception could contain any hint as to why it crashed out I kept it. Happy to remove if you think redundant

Reply:
Ahh right, I missed the first argument in the method call (`e`). Now it makes more sense 👍